sklearn/metrics/scorer.py

```diff
@@ -66,6 +67,34 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def _get_labels_from_estimator(self, estimator, target_type):
+        """Check if estimator has a classes argument and if scorer has a
+        labels argument and return new kwargs for scorer if label not already
+        set in make score initialization"""
+        est_has_classes = hasattr(estimator, "classes_")
+        scorer_has_labels = "labels" in signature(
```

Comment:
to your first point, i'm not sure how that'll be an issue. since we are defining this code inside BaseScorer, it will only be executed if the user uses make_scorer to create a custom metric. if they define their own class, then its upto them how they handle passing labels.

the copy_classes idea sounds good. we can actually keep it a string so that the user can enter any other parameter name as well where they want the classes to be copied. if not, they an just pass None. sounds right?